File: java/src/main/java/org/urlshortener/dal/repositories/user/UserRepository.java
package org.urlshortener.dal.repositories.user;

import lombok.NonNull;
import org.urlshortener.dal.entities.User;

import javax.validation.Valid;

public interface UserRepository {
    User merge(@Valid User user);

    void deleteById(long id);

    User findByName(@NonNull String name);

    User findById(long userId);

    boolean existsByName(@NonNull String name);
}

File: kotlin/src/main/kotlin/com/urlshortener/dal/repositories/UserRepository.kt
package com.urlshortener.dal.repositories

import com.urlshortener.dal.entities.User
import javax.validation.Valid

interface UserRepository {
    fun merge(@Valid user: User): User
    fun deleteById(id: Long)
    fun findByName(name: String): User?
    fun findById(userId: Long): User?
    fun existsByName(name: String): Boolean
}